### PR TITLE
Move deployment config to etcdgeneric storage patterns

### PR DESCRIPTION
**Pull request by derekwaynecarr**
This moves the storage of ```DeploymentConfig``` objects to follow latest upstream patterns.

/cc @ironcladlou @abhgupta @smarterclayton

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Related to https://github.com/openshift/origin/issues/2104

File: pkg/deploy/registry/deployconfig/strategy.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+// AllowCreateOnUpdate is false for DeploymentConfig objects.
+func (strategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// PrepareForCreate clears fields that are not allowed to be set by end users on creation.
+func (strategy) PrepareForCreate(obj runtime.Object) {
+	_ = obj.(*api.DeploymentConfig)
```
Comment (smarterclayton) [inline pkg/deploy/registry/deployconfig/strategy.go]: We need a follow up issue to determine whether PrepareForCreate/Update should prevent status.latestVersion from being set out of order.  There are advantages and disadvantages.  Maybe a todo.

File: pkg/deploy/registry/deployconfig/strategy.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+// Validate validates a new policy.
+func (strategy) Validate(ctx kapi.Context, obj runtime.Object) fielderrors.ValidationErrorList {
+	return validation.ValidateDeploymentConfig(obj.(*api.DeploymentConfig))
+}
+
+// ValidateUpdate is the default update validation for an end user.
+func (strategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
+	// TODO: distinguish updates from create, for now, this is preserving old implementation behavior
+	return validation.ValidateDeploymentConfig(obj.(*api.DeploymentConfig))
```
Comment (smarterclayton) [inline pkg/deploy/registry/deployconfig/strategy.go]: We must explicitly validate ObjectMeta update.  Otherwise someone can change the UID.

**Comment by derekwaynecarr** [comment]
Then builds have the same problem.  Will update

**Comment by derekwaynecarr** [comment]
Is there any need to keep /deployments and the storage code laying around.  It's confusing and would like to purge it as well.

**Comment by smarterclayton** [comment]
No, you can nuke it.

----- Original Message -----

**Comment by smarterclayton** [comment]
Other than objectmeta update validation this is LGTM

**Comment by derekwaynecarr** [comment]
Got side tracked on doc stuff, will nuke deployments, fix update, and then look to merge.

**Comment by derekwaynecarr** [comment]
Fixed validation, will kill /deployments code in a later PR so I can keep this merge simpler.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**